lib/devices/android/selendroid.js

```diff
@@ -220,32 +220,24 @@ Selendroid.prototype.checkInternetPermissionForApp = function (cb) {
 };
 
 Selendroid.prototype.checkSelendroidCerts = function (cb) {
-  var alreadyReturned = false
-    , checks = 0;
-
-  var onDoneSigning = function () {
-    checks++;
-    if (checks === 2 && !alreadyReturned) {
-      cb();
-    }
-  };
-
-  // these run in parallel
   var apks = [this.selendroidServerPath, this.args.app];
-  _.each(apks, function (apk) {
+    
+  async.eachSeries(apks, function(apk, done){
     logger.debug("Checking signed status of " + apk);
     this.adb.checkApkCert(apk, this.args.appPackage, function (err, isSigned) {
-      if (err) return cb(err);
-      if (isSigned) return onDoneSigning();
+      if (err) return done(err);
+      if (isSigned) return done();
       this.adb.sign(apk, function (err) {
-        if (err && !alreadyReturned) {
-          alreadyReturned = true;
-          return cb(err);
-        }
-        onDoneSigning();
+        if (err) return done(err);
+		done();
```

Comment:
@moizjv : my local file has right indentation. I don't know why its not picking same in this online version (I am using bracket editor) and same with following comment. Let me see if i can do one more commit